internal/cli/alpha/content/terraform.go

```diff
@@ -0,0 +1,368 @@
+	if diags.Err() != nil {
+		return nil, errors.Wrap(diags.Err(), "while loading Terraform module")
+	}
+
+	var (
+		interfacePath     = cfg.InterfacePathWithRevision
+		interfaceRevision = "0.1.0"
+	)
+
+	pathSlice := strings.Split(cfg.InterfacePathWithRevision, ":")
+	if len(pathSlice) == 2 {
+		interfacePath = pathSlice[0]
+		interfaceRevision = pathSlice[1]
+	}
+
+	input := &terraformTemplatingInput{
+		templatingInput: templatingInput{
+			Name:      cfg.ManifestName,
+			Prefix:    cfg.ManifestsPrefix,
+			Revision:  cfg.ManifestRevision,
+			Variables: make([]inputVariable, 0, len(module.Variables)),
+		},
+		InterfacePath:     interfacePath,
+		InterfaceRevision: interfaceRevision,
+		ModuleSourceURL:   cfg.ModuleSourceURL,
+		Outputs:           make([]outputVariable, 0, len(module.Outputs)),
+		Provider:          cfg.Provider,
+	}
+
+	for _, tfVar := range module.Variables {
+		// Skip default for now, as there are problems, when it is a multiline string or with doublequotes in it.
+		input.Variables = append(input.Variables, inputVariable{
+			Name:        tfVar.Name,
+			Type:        getTypeFromTerraformType(tfVar.Type),
+			Description: tfVar.Description,
+		})
+	}
+
+	sort.Slice(input.Variables, func(i, j int) bool {
+		return input.Variables[i].Name < input.Variables[j].Name
+	})
+
+	for _, tfOut := range module.Outputs {
+		input.Outputs = append(input.Outputs, outputVariable{
+			Name: tfOut.Name,
+		})
+	}
+
+	sort.Slice(input.Outputs, func(i, j int) bool {
+		return input.Outputs[i].Name < input.Outputs[j].Name
+	})
+
+	return input, nil
+}
+
+// Terraform types: https://www.terraform.io/docs/language/expressions/types.html
+func getTypeFromTerraformType(t string) string {
+	if strings.HasPrefix(t, "list") || strings.HasPrefix(t, "tuple") {
+		return "array"
+	}
+
+	switch t {
+	case "string":
+		return "string"
+	case "number":
+		return "number"
+	case "bool":
+		return "boolean"
+	case "null":
+		return "null"
+	}
+
+	return "object"
+}
+
+const (
+	terraformImplementationManifestTemplate = `ocfVersion: 0.0.1
+revision: {{ .Revision }}
+kind: Implementation
+metadata:
+  prefix: "cap.implementation.{{ .Prefix }}"
+  name: {{ .Name }}
+  displayName: "{{ .Name }} Action"
+  description: "{{ .Name }} Action"
+  documentationURL: https://example.com
+  supportURL: https://example.com
+  maintainers:
+    - email: dev@example.com
+      name: Example Dev
+      url: https://example.com
+  license:
+    name: "Apache 2.0"
+
+spec:
+  appVersion: "1.0.x" # Set the supported application version here
+  additionalInput:
+    parameters:
+      typeRef:
+        path: "cap.type.{{ .Prefix }}.{{ .Name }}-input"
+        revision: 0.1.0
+
+  outputTypeInstanceRelations:
+    config:
+      uses:
+        - terraform-release
+
+  implements:
+    - path: {{if .InterfacePath}}cap.interface.{{ .InterfacePath }}{{else}}"cap.interface..." # Put here the path of the implemented Interface{{end}}
```

Comment:
It would be easier to find places which user should edit if you add `TODO:` prefix to all your comments that require action from user.

Reply:
I thought about it, but it could be a bit misleading, if this TODO is code related or for the Content Creator.